From 1fbff0e24237dfd4e94667db045a648d31f34fa2 Mon Sep 17 00:00:00 2001
From: Gerard Andre <gerard.andre@intel.com>
Date: Thu, 12 Mar 2015 15:05:40 +0100
Subject: [PATCH 2/3] gc2335: Adjust MIPI clock frequency, and soc mipi
 settle_cnt

There should be some problem with GC2235 PLL circuit
to have MIPI@144MHz,to cause stability issue on
output MIPI signal.
MIPI@120MHz is verified by the vender.
After adjusted the MIPI frequency, we need to adjust SOC.
MIPI settle_cnt from 72ns to 100ns to match this change.

Change-Id: I6e8228f2175d12c96ae0acd877d776693ffbfd65
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-26557
Signed-off-by: liujianjun <jianjun.liu@intel.com>
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/46230
---
 drivers/external_drivers/camera/drivers/media/i2c/gc2235.h              | 2 +-
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
index 5ef9894..4431e0c 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
@@ -237,7 +237,7 @@ static struct gc2235_reg const gc2235_init_settings[] = {
 	{ GC2235_8BIT, 0xf6, 0x00 },
 	{ GC2235_8BIT, 0xfc, 0x06 },
 	{ GC2235_8BIT, 0xf7, 0x15 },
-	{ GC2235_8BIT, 0xf8, 0x85 },
+	{ GC2235_8BIT, 0xf8, 0x84 },
 	{ GC2235_8BIT, 0xf9, 0xfe },
 	{ GC2235_8BIT, 0xfa, 0x00 },
 	{ GC2235_8BIT, 0xfe, 0x00 },
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c
index fd563ef..574fbe3 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c
@@ -337,7 +337,7 @@ static void atomisp_csi2_configure_isp2401(struct atomisp_sub_device *asd)
 	static const short int coeff_dat_termen[] = { 0, 0 };
 	static const short int coeff_dat_settle[] = { 85, -2 };
 	static const int TERMEN_DEFAULT		  = 0 * 0;
-	static const int SETTLE_DEFAULT		  = 0x480;
+	static const int SETTLE_DEFAULT		  = 0x640;
 	static const hrt_address csi2_port_base[] = {
 		[ATOMISP_CAMERA_PORT_PRIMARY]     = CSI2_PORT_A_BASE,
 		[ATOMISP_CAMERA_PORT_SECONDARY]   = CSI2_PORT_B_BASE,
-- 
1.9.1

